cmd/lncli/cmd_payments.go

```diff
@@ -1394,6 +1394,123 @@ func buildRoute(ctx *cli.Context) error {
 	return nil
 }
 
+var deletePaymentsCommand = cli.Command{
+	Name:     "deletepayments",
+	Category: "Payments",
+	Usage:    "Delete a single or multiple payments from the database.",
+	Description: `
+	By default this command deletes all failed payments from the database to
+	reclaim disk space.
+
+	If so desired, _ALL_ payments (even the successful ones) can be deleted
+	by specifying --include_non_failed.
+
+	If a --payment_hash is specified, that single payment is deleted,
+	independent of its state.
+
+	If --failed_htlcs_only is specified then the payments themselves (or the
+	single payment itself if used with --payment_hash) is not deleted, only
+	the information about any failed HTLC attempts during the payment.
+
+	NOTE: Removing payments from the database does free up disk space within
+	the internal bbolt database. But that disk space is only reclaimed after
+	compacting the database. Users might want to turn on auto compaction
+	(db.bolt.auto-compact=true in the config file or --db.bolt.auto-compact
+	as a command line flag) and restart lnd after deleting a large number of
+	payments to see a reduction in the file size of the channel.db file.
+	`,
+	Action: actionDecorator(deletePayments),
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name: "payment_hash",
```

Comment:
How about requiring the `-all` flag like I did in (https://github.com/lightningnetwork/lnd/pull/5778) to make this harder to use incorrectly? So just typing `lncli deletepayments` returns with the help, and forces you to understand what's going on and pick an option.

Reply:
Yeah, good idea. Done :+1: